### PR TITLE
protected context area

### PR DESCRIPTION
**Pull request by j8seangel**
MPA as context area example:
image

Also, restore dataviewInstance on `+` icon button click in context areas and heatmap after removal all of them

### CODE REVIEW

File: applications/fishing-map/src/features/sidebar/context-areas/ContextAreaLayerPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  const layerActive = dataview?.config?.visible ?? true
+  const onToggleLayerActive = () => {
+    upsertDataviewInstance({
+      id: dataview.id,
+      config: { visible: !layerActive, color },
+    })
+  }
+
+  const color = dataview?.config?.color
+  const changeTrackColor = (color: ColorBarOption) => {
```
Comment (nerik) [inline applications/fishing-map/src/features/sidebar/context-areas/ContextAreaLayerPanel.tsx:38]: nitpick
```suggestion
  const changeColor = (color: ColorBarOption) => {
```

File: applications/fishing-map/src/features/sidebar/context-areas/ContextAreaSection.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import React from 'react'
+import { useSelector } from 'react-redux'
+import { selectContextAreasDataviews } from 'features/workspace/workspace.selectors'
+import styles from 'features/sidebar/common/Sections.module.css'
+import LayerPanel from './ContextAreaLayerPanel'
+
+function VesselsSection(): React.ReactElement {
```
Comment (nerik) [inline applications/fishing-map/src/features/sidebar/context-areas/ContextAreaSection.tsx:7]: nitpick
```suggestion
export default function(): React.ReactElement {
```

File: applications/fishing-map/src/features/sidebar/heatmaps/HeatmapsSection.tsx [hunk 3]
```diff
@@ -1,17 +1,29 @@
...
 import { selectTemporalgridDataviews } from 'features/workspace/workspace.selectors'
 import styles from 'features/sidebar/common/Sections.module.css'
+import { useDataviewInstancesConnect } from 'features/workspace/workspace.hook'
 import LayerPanel from './HeatmapLayerPanel'
 
 function HeatmapsSection(): React.ReactElement {
   const dataviews = useSelector(selectTemporalgridDataviews)
+  const { removeDataviewInstance } = useDataviewInstancesConnect()
+  const onAddClick = useCallback(() => {
+    removeDataviewInstance('fishing-1')
```
Comment (nerik) [inline applications/fishing-map/src/features/sidebar/heatmaps/HeatmapsSection.tsx:13]: Unclear why `onAddSomething` triggers `removeSomething`.
Maybe `toggleDataviewInstance` ?

**Review by nerik — CHANGES_REQUESTED** [review]
Looks good 👍 
I have a bunch of comments about naming. Sometimes maybe coming from some copy-pasted code. It's just nitpick but probably better to fix that now to avoid future WTF moments.
In the same spirit: the UserContext generator is deifinitely a good base for context layers, it would make sense to rename it simply Context. Adding that to the backlog + add interaction

File: applications/fishing-map/src/features/sidebar/heatmaps/HeatmapsSection.tsx (hunk 3 above)
Comment (j8seangel) [reply applications/fishing-map/src/features/sidebar/heatmaps/HeatmapsSection.tsx:13]: For now, this is just a workaround that will be deleted later but something we need to discuss is how to remove an existing dataviewInstance from the workspace...

**Comment by j8seangel** [comment]
Thanks for the naming catches, I agree with all of them except in:

> In the same spirit: the UserContext generator is definitely a good base for context layers, it would make sense to rename it simply Context. Adding that to the backlog + add interaction

it is done because this was the way to have it working easily but we'll need to keep user-context as simple as it is, for now, and create a new one with interaction, different styles for the EEZ, etc... but added a TODO to not forget it

File: applications/fishing-map/src/features/sidebar/context-areas/ContextAreaSection.tsx (hunk 2 above)
Comment (j8seangel) [reply applications/fishing-map/src/features/sidebar/context-areas/ContextAreaSection.tsx:7]: I prefer having the name so in the Component debugger tree it is easily to debug rathern than `anonymous function`

Comment (nerik) [reply applications/fishing-map/src/features/sidebar/context-areas/ContextAreaSection.tsx:7]: I was bothered by the fact it's called 'Vessels'

File: applications/fishing-map/src/features/sidebar/heatmaps/HeatmapsSection.tsx (hunk 3 above)
Comment (nerik) [reply applications/fishing-map/src/features/sidebar/heatmaps/HeatmapsSection.tsx:13]: Please add a comment/TODO.

**Pull request merged**